tests/view_tests/tests/test_debug.py

```diff
@@ -403,6 +408,33 @@ def test_reporting_frames_without_source(self):
         text = reporter.get_traceback_text()
         self.assertIn('"generated" in funcName', text)
 
+    def test_reporting_frames_for_cyclic_reference(self):
+        try:
+            def test_func():
+                try:
+                    raise RuntimeError('outer') from RuntimeError('inner')
+                except RuntimeError as exc:
+                    raise exc.__cause__
+            test_func()
+        except Exception:
+            exc_type, exc_value, tb = sys.exc_info()
+        request = self.rf.get('/test_view/')
+        reporter = ExceptionReporter(request, exc_type, exc_value, tb)
+
+        def generate_traceback_frames(*args, **kwargs):
+            nonlocal tb_frames
+            tb_frames = reporter.get_traceback_frames()
+        tb_frames = None
+        tb_generator = threading.Thread(target=generate_traceback_frames)
+        tb_generator.start()
+        tb_generator.join(timeout=1)
```

Comment:
I think this could be increased to 10 just in case the system is really slow. The important thing is that it doesn't loop forever.